runhouse/servers/cluster_servlet.py

```diff
@@ -217,3 +234,130 @@ async def aclear_all_references_to_env_servlet_name(self, env_servlet_name: str)
+                )
+                await asyncio.sleep(3600)
+            finally:
+
+                await asyncio.sleep(interval_size)
+
+    def send_status_info_to_den_sync(self):
+        asyncio.run(self.asend_status_info_to_den())
+
+    def schedule_post_status(self):
+        # adding post status to den thread
+        logger.debug("adding send_status_info_to_den_sync to thread pool")
+        # delay the start of post_status_thread, so we'll finish the cluster startup properly
+        post_status_thread = threading.Timer(
+            STATUS_CHECK_DELAY, self.send_status_info_to_den_sync
+        )
+        logger.debug("starting send_status_info_to_den_sync thread")
+
+        post_status_thread.start()
+
+    def set_obj_store(self, associated_obj_store):
+        self.obj_store = associated_obj_store
+
+    # TODO [SB]: remove this method in PR 806
+    async def astatus_cluster_servlet(self):
+        if not self.obj_store:
+            logger.error(
+                "The relevant obj-store is not associated with the cluster servlet, therefore can't get cluster status."
+                " Setting the obj_store can be done using cluster_servlet.set_obj_store(associated_obj_store)."
+            )
+            raise ClusterServletError(
+                "Can't find the obj_store associated with current ClusterServlet."
+            )
+        return await self.obj_store.astatus()
+
+    async def astatus(self):
+        import psutil
+
+        from runhouse.utils import get_pid
+
+        config_cluster = copy.deepcopy(self.cluster_config)
+
+        # poping out creds because we don't want to show them in the status
+        config_cluster.pop("creds", None)
+
+        # getting cluster servlets (envs) and their related objects
+        cluster_servlets = {}
+        cluster_envs_env_utilization_data = {}
+        for env in self._initialized_env_servlet_names:
+            try:
+                (
+                    objects_in_env_modified,
+                    env_utilization_data,
+                ) = await obj_store.acall_actor_method(
+                    obj_store.get_env_servlet(env), method="status_local"
+                )
+                cluster_servlets[env] = objects_in_env_modified
+                cluster_envs_env_utilization_data[env] = env_utilization_data
+            except ObjStoreError:
+                cluster_servlets[env] = []
+                cluster_envs_env_utilization_data[env] = {}
+        config_cluster["envs"] = cluster_servlets
+        config_cluster["server_pid"] = get_pid()
+        config_cluster["runhouse_version"] = runhouse.__version__
+
+        # TODO: decide if we need this info at all: cpu_usage, memory_usage, disk_usage
+        cpu_usage = psutil.cpu_percent(interval=1)
+
+        # Fields: `available`, `percent`, `used`, `free`, `active`, `inactive`, `buffers`, `cached`, `shared`, `slab`
+        memory_usage = psutil.virtual_memory()._asdict()
```

Comment:
maybe we can even enforce that these fields are there? by putting them into a model or something?